tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py

```diff
@@ -108,107 +63,144 @@ def parameters(cls):
 
-        map_coco_img_file_name_to_img_id = {os.path.basename(v['file_name']): v['id'] for v in coco.dataset['images']}
-        assert len(map_coco_img_file_name_to_img_id) == len(coco.dataset['images']), "Image name duplications"
+        coco_annotation = self._create_json(coco_data_to_store)
+        coco = COCO(str(coco_annotation))
 
-        return coco, map_coco_img_file_name_to_img_id, map_pred_label_id_to_coco_cat_id
+        return coco
 
-    @staticmethod
-    def _convert_data_to_coco_format(
-            predictions, map_coco_img_file_name_to_img_id, map_pred_label_id_to_coco_cat_id, iou_type='bbox'
-    ):
+    def _convert_data_to_coco_format(self, predictions):
         coco_data_to_store = []
         for pred in predictions:
             prediction_data_to_store = []
-            cur_name = pred.identifier
-            cur_name = os.path.basename(cur_name)
-            assert cur_name in map_coco_img_file_name_to_img_id
-            cur_img_id = map_coco_img_file_name_to_img_id[cur_name]
+            cur_name = pathlib.PurePath(pred.identifier).name
+            cur_img_id = int(cur_name.split(".")[0])
 
             labels = pred.labels.tolist()
             scores = pred.scores.tolist()
             cur_num = len(labels)
             assert len(scores) == cur_num
 
-            coco_cats = [map_pred_label_id_to_coco_cat_id[lbl] for lbl in labels]
-            for (s, cur_cat) in zip(scores, coco_cats):
+            for (s, cur_cat) in zip(scores, labels):
                 prediction_data_to_store.append({
                     'image_id': cur_img_id,
                     'score': s,
                     'category_id': cur_cat,
                     '_image_name_from_dataset': cur_name,
                 })
-            iou_specific_converter = iou_specific_processing.get(iou_type)
-            if iou_specific_converter is None:
-                raise ValueError("unknown iou type: '{}'".format(iou_type))
-            prediction_data_to_store = iou_specific_converter(prediction_data_to_store, pred)
+            prediction_data_to_store = self._iou_type_data_to_coo(prediction_data_to_store, pred)
             coco_data_to_store.extend(prediction_data_to_store)
 
         return coco_data_to_store
 
+    def _iou_type_specific_coco_annotation(self, annotation_data_to_store, annotation):
+        annotation_data_to_store = self._iou_type_data_to_coo(annotation_data_to_store, annotation)
+        return annotation_data_to_store
+
+    def _prepare_data_for_annotation_file(
+            self, annotations, dataset_label_map
+    ):
+        coco_annotation_to_store = []
+        coco_category_to_store = []
+        coco_image_to_store = []
+        label_map = set()
+        count = 0
+        for annotation in annotations:
+            annotation_data_to_store = []
+            cur_name = pathlib.PurePath(annotation.identifier).name
+            cur_img_id = int(cur_name.split(".")[0])
+
+            if self.iou_type != 'segm':
```

Comment:
could you please explain these changes?

Reply:
annotation.labels is ndarray in case iou_type is keypoint and bbox and annotation.labels is list in case iou_type is segm

Reply:
I think that condition can be removed , but following changes need to be added:
`           labels = annotation.labels

            if annotation.metadata.get('iscrowd'):
                iscrowds = annotation.metadata.get('iscrowd')
            else:
                iscrowds = [0]*annotation.size

            for cur_cat, iscrowd in zip(labels, iscrowds):
                annotation_data_to_store.append({
                    'id': count,
                    'image_id': cur_img_id,
                    'category_id': int(cur_cat),
                    '_image_name_from_dataset': cur_name,
                    'iscrowd': iscrowd
                })
                count += 1
                label_map.add(int(cur_cat))`

Reply:
please review again